### ISSUE TITLE
Geocoding not working with ArcGIS Enterprise Portal

### ISSUE THREAD

**Issue by jrmatchett** [Esri/arcgis-python-api#1108]
**Describe the bug**
The `geocode` function produces an error when using an Enterprise Portal `GIS` object. Geocoding works fine with an ArcGIS Online `GIS` object.

**To Reproduce**
```python
from arcgis import GIS
from arcgis.geocoding import geocode

gis = GIS('https://maps.myenterprise.org/portal', client_id='XXXX')

geocode('380 New York St, Redlands, CA 92373')
```
error:
```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/var/folders/2j/yrfvl2bj1ld05s_r3gwk39yh0000gp/T/ipykernel_33164/2847291716.py in <module>
----> 1 geocode('380 New York St, Redlands, CA 92373')

~/opt/miniconda3/envs/arcgis191/lib/python3.8/site-packages/arcgis/geocoding/_functions.py in geocode(address, search_extent, location, distance, out_sr, category, out_fields, max_locations, magic_key, for_storage, geocoder, as_featureset, match_out_of_range, location_type, lang_code, source_country)
   1333     if geocoder is None:
   1334         geocoder = arcgis.env.active_gis._tools.geocoders[0]
-> 1335     return geocoder._geocode(
   1336         address,
   1337         search_extent,

~/opt/miniconda3/envs/arcgis191/lib/python3.8/site-packages/arcgis/geocoding/_functions.py in _geocode(self, address, search_extent, location, distance, out_sr, category, out_fields, max_locations, magic_key, for_storage, as_featureset, match_out_of_range, location_type, lang_code, source_country)
    223         if address is not None:
    224             if isinstance(address, str):
--> 225                 params[self._address_field] = address
    226             elif isinstance(address, dict):
    227                 params.update(address)

AttributeError: 'Geocoder' object has no attribute '_address_field'
```

**Platform (please complete the following information):**
 - OS: macOS 11.6
 - Browser: Safari and Chrome
 - Python API Versions: Tested on `1.8.5`, `1.9.0`, and `1.9.1`.

**Comment by jrmatchett** [comment]
Digging into this a bit further, the `Geocoder`'s `_address_field` isn't getting set because a token isn't being generated when attempting to retrieve its properties (in this case, it's an Enterprise Portal geocoding service that isn't publicly accessible). The root cause appears to be what I pointed out in [this issue](https://github.com/Esri/arcgis-python-api/issues/776#issuecomment-743791676) with the `Connection.generate_portal_server_token` method not handing oauth authentication.

**Comment by achapkowski** [comment]
The issue is resolved in v2.0.0 of the Python API, which will come out in Q1 of 2022

**Comment by thakkarsmit11** [comment]
Hi, I am having same issue. Is this issue resolved? December 2023
